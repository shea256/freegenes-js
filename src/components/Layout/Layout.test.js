/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import App from '../App';
import Layout from './Layout';

// const middlewares = []
const mockStore = configureMockStore();

describe('Layout', () => {
  test('renders children correctly', () => {
    const store = mockStore({});
    const wrapper = renderer
      .create(
        <App
          context={{
            store,
            insertCss: () => {},
            fetch: () => {},
            pathname: '',
          }}
        >
          <Layout store={store}>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
