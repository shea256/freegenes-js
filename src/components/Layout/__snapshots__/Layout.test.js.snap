// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Renders children correctly 1`] = `
<div>
  <div>
    <div>
      <nav
        className="navbar navbar-expand-md navbar-dark bg-dark"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          FreeGenes
        </a>
        <button
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="ml-auto navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/about"
                onClick={[Function]}
              >
                About
              </a>
            </li>
            <li
              className="dropdown nav-item"
              onKeyDown={[Function]}
            >
              <a
                aria-haspopup={true}
                className="dropdown-toggle nav-link"
                href="#"
                onClick={[Function]}
              >
                Resources
              </a>
              <div
                aria-hidden={true}
                className="dropdown-menu dropdown-menu-right"
                role="menu"
                tabIndex="-1"
              >
                <a
                  className="dropdown-item"
                  href="/collections"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                >
                  Biopart Collections
                </a>
                <a
                  className="dropdown-item"
                  href="/parts"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                >
                  Bioparts
                </a>
                <a
                  className="dropdown-item"
                  href="/authors"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                >
                  Biopart Authors
                </a>
                <a
                  className="dropdown-item"
                  href="/plates"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                >
                  Plates
                </a>
                <a
                  className="dropdown-item"
                  href="/wells"
                  onClick={[Function]}
                  role="menuitem"
                  tabIndex="0"
                >
                  Plate Wells
                </a>
              </div>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="child"
  />
  <div
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © FreeGenes
      </span>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <span
        className="spacer"
      >
        ·
      </span>
      <a
        className="link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </div>
  </div>
</div>
`;
